torch/_inductor/codecache.py

```diff
@@ -379,6 +386,254 @@ def write_atomic(path: str, content: Union[str, bytes]) -> None:
     tmp_path.rename(path)
 
 
+@dataclasses.dataclass
+class TensorMetadata:
+    """
+    The Tensor metadata relevant when hashing FxGraph cache keys.
+    """
+
+    dtype: torch.dtype
+    shape: torch.Size
+    stride: Tuple[Any, ...]
+    device: torch.device
+    requires_grad: bool
+    storage_offset: int
+    memory_format: Optional[torch.memory_format]
+    is_quantized: bool
+
+
+@dataclasses.dataclass
+class TensorMetadataAndValues:
+    """
+    TensorMetadata plus the elements as a list of raw values.
+    """
+
+    tensor_metadata: TensorMetadata
+    values: List[Any]
+
+
+class DynamicShapeError(BaseException):
+    """
+    TODO(masnesral): Support dynamic shapes and remove this.
+    """
+
+    pass
+
+
+def extract_tensor_metadata(t: torch.Tensor) -> TensorMetadata:
+    """
+    Extract the TensorMetadata of a tensor.
+    """
+    if not all(isinstance(e, int) for e in t.shape):
+        raise DynamicShapeError()
+
+    # Note the order here matters since is_contiguous() can be True
+    # for more than one memory format.
+    memory_formats = [
+        torch.contiguous_format,
+        torch.channels_last,
+        torch.channels_last_3d,
+    ]
+
+    memory_format = None
+    for fmt in memory_formats:
```

Comment:
you can replace this loop with `torch._prims_common.suggest_memory_format`, although it will default to contiguous, so maybe after invoking that api check `suggested_memory_format = suggested_memory_format if t.is_contiguous(suggested_memory_format) else None`